biogtr/datasets/sleap_dataset.py

```diff
@@ -284,25 +301,35 @@ def get_instances(self, label_idx: List[int], frame_idx: List[int]) -> list[dict
                 else:
                     anchors = self.anchors
 
-                for anchor in anchors:
-                    if anchor == "midpoint" or anchor == "centroid":
-                        centroid = np.nanmean(np.array(list(pose.values())), axis=0)
+                anchors_to_drop = np.random.permutation(anchors)
+                anchor_dropout_p = np.random.uniform(size=len(anchors_to_drop))
+                dropped_anchor_inds = np.where(
+                    anchor_dropout_p < self.node_dropout["p"]
+                )
+                anchor_dropout_p = anchor_dropout_p[dropped_anchor_inds]
+                n_anchors_to_drop = min(self.node_dropout["n"], len(anchor_dropout_p))
+                dropped_anchor_inds = np.argpartition(
+                    anchor_dropout_p, -n_anchors_to_drop
+                )[-n_anchors_to_drop:]
+                dropped_anchors = anchors_to_drop[dropped_anchor_inds]
```

Comment:
This is all a bit wild and should be in its own, very heavily tested function.

There are a lot of vectorized conditionals going on and I'm not super clear on the edge case behaviors.